src/controllers/ride/removeRequest.ts

```diff
@@ -60,6 +60,9 @@ export const removeRequest = async (req: Request, res: Response) => {
       return res.status(404).json({ message: "Ride not found in the DB." });
     }
 
+    if (reqUserEmail == userEmail)
```

Comment:
Umm the if condn should also make sure `reqEmail` is `rideObj.originalPoster.email`